src/spring-cloud/azext_spring_cloud/_resource_quantity.py

```diff
@@ -0,0 +1,34 @@
+import re
+from knack.util import CLIError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+def validate_cpu(cpu):
+    if cpu is None:
+        return None
+
+    if not re.match(r"^\d+m?$", cpu):
+        raise CLIError("CPU quantity should be millis (500m) or integer (1, 2, ...)")
+
+    return cpu
+
+
+def validate_memory(memory):
+    if memory is None:
+        return None
+
+    unified = memory
+    try:
+        # For backward compatibility, to convert integer value to value with Gi unit
+        int(memory)
+        logger.warning("Memory quantity [--memory] should be specified with unit, such as 512Mi, 1Gi. "
+                       "Support for integer quantity will be dropped in future release.")
+        unified = memory + "Gi"
+    except ValueError:
+        pass
+
+    if not re.match(r"^\d+[MG]i$", unified):
```

Comment:
Could you add some explanation for the function of the regular expression?

Reply:
We allow the user to specify the memory quantity with either `###Mi` or `###Gi`, for example, `1024Mi`, `3Gi`. The code above this line is to add backward compatibility for legacy input, which uses integer for `Gi` values.

The regular expression here add some client side validation for the user input, to restrict it to be some digits followed by `Mi` or `Gi`. (Note that integer input will be converted to `###Gi` before this validation).